### PR TITLE
docs(recommendations): update documentation for recommendations

### PR DESCRIPTION
**Pull request by shortcuts**
**Summary**

- Updates the docs for `react-recommendations`
- Adds doc for `js-recommendations`

### CODE REVIEW

File: packages/js-recommendations/README.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+```html
+<div id="relatedProducts"></div>
+```
+
+Then, inject results into it by calling the `relatedProducts` function and providing the [`container`](#container). It can be a [CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors) or an [Element](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement).
+
+```js
+/** @jsx h */
+
```
Comment (francoischalifour) [inline packages/js-recommendations/README.md]:
```suggestion
/** @jsx h */
```

File: packages/js-recommendations/README.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+import { relatedProducts } from '@algolia/js-recommendations';
+import algoliasearch from 'algoliasearch';
+
+const appId = 'HYDY1KWTWB';
+const apiKey = '28cf6d38411215e2eef188e635216508';
+const indexName = 'gstar_demo_test';
+
+const searchClient = algoliasearch(appId, apiKey);
+
+function renderRelatedProducts({ currentObjectID }) {
```
Comment (francoischalifour) [inline packages/js-recommendations/README.md]: We can remove this function abstraction from the code snippet.

File: packages/js-recommendations/README.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        </pre>
+      );
+    },
+  });
+}
+```
+
+#### Props
+
+The component accepts all the [shared props](#shared-props) and the following:
```
Comment (francoischalifour) [inline packages/js-recommendations/README.md]:
```suggestion
The function accepts all the [shared props](#shared-props) and the following:
```

File: packages/js-recommendations/README.md [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+##### `translations`
+
+> `Translations`
+
+```ts
+type RecommendationTranslations = Partial<{
+  title: string;
+  showMore: string;
+}>;
+```
```
Comment (francoischalifour) [inline packages/js-recommendations/README.md]: We can wrap this in `details`.

File: packages/react-recommendations/README.md [hunk 5]
```diff
@@ -65,17 +65,93 @@ The product component to display.
 > `Translations`
```
Comment (francoischalifour) [inline packages/react-recommendations/README.md]:
```suggestion
> `RecommendationTranslations`
```

File: packages/react-recommendations/README.md [hunk 6]
```diff
@@ -176,17 +252,93 @@ The product component to display.
...
```
Comment (francoischalifour) [inline packages/react-recommendations/README.md]:
```suggestion
> `RecommendationTranslations`
```

File: packages/react-recommendations/README.md [hunk 7]
```diff
@@ -176,17 +252,93 @@ The product component to display.
...
+
+```ts
+type ViewProps<
+  TItem extends RecordWithObjectID,
+  TTranslations extends Record<string, string>,
+  TClassNames extends Record<string, string>
+> = {
+  items: TItem[];
+  itemComponent({ item: TItem }): JSX.Element;
+  classNames?: Partial<TClassNames>;
```
Comment (francoischalifour) [inline packages/react-recommendations/README.md]: We don't pass `classNames`.

Comment (francoischalifour) [inline packages/react-recommendations/README.md]: We can remove the `Partial`s here.

File: packages/react-recommendations/README.md [hunk 8]
```diff
@@ -326,4 +478,22 @@ Additional filters to use as fallback should there not be enough recommendations
...
+
+```ts
+type RecordWithObjectID<TItem> = TItem & {
+  objectID: string;
+};
+```
+
+</details>
+
+Receives the items, and is called before displaying them. Should return a new array with the same shape as the original array. Useful for mapping over the items to transform, and remove or reorder them.
```
Comment (francoischalifour) [inline packages/react-recommendations/README.md]: We can move this function above `details`.

```suggestion
Function to transform the items retrieved by Algolia. It's useful to edit, add, remove or reorder them.
```

File: packages/js-recommendations/README.md (hunk 1 above)
Comment (shortcuts) [reply packages/js-recommendations/README.md]: Did you meant it needs to be removed?

File: packages/js-recommendations/README.md (hunk 2 above)
Comment (shortcuts) [reply packages/js-recommendations/README.md]: Then `currentObjectID` becomes "undefined", should we rename it to `selectedObjectID` so it's explicit?

File: packages/react-recommendations/README.md (hunk 7 above)
Comment (shortcuts) [reply packages/react-recommendations/README.md]: Make sense

File: packages/js-recommendations/README.md (hunk 4 above)
Comment (shortcuts) [reply packages/js-recommendations/README.md]: Done in https://github.com/algolia/ui-components/pull/18/commits/a1991cfb5cb8bc7bd4502425134439493283c578

File: packages/react-recommendations/README.md (hunk 5 above)
Comment (shortcuts) [reply packages/react-recommendations/README.md]: Done in https://github.com/algolia/ui-components/pull/18/commits/a1991cfb5cb8bc7bd4502425134439493283c578

File: packages/react-recommendations/README.md (hunk 6 above)
Comment (shortcuts) [reply packages/react-recommendations/README.md]: Done in https://github.com/algolia/ui-components/pull/18/commits/a1991cfb5cb8bc7bd4502425134439493283c578

File: packages/react-recommendations/README.md (hunk 7 above)
Comment (shortcuts) [reply packages/react-recommendations/README.md]: Done in https://github.com/algolia/ui-components/pull/18/commits/25fa8410a96e0521a21ab5738b56d938434a2622

Comment (shortcuts) [reply packages/react-recommendations/README.md]: Done in https://github.com/algolia/ui-components/pull/18/commits/0e8e81363ba4baa6c642e47992f3f1ce29391dfb

File: packages/react-recommendations/README.md (hunk 8 above)
Comment (shortcuts) [reply packages/react-recommendations/README.md]: Done in https://github.com/algolia/ui-components/pull/18/commits/3335abd6b66e9621bb5985eb477a8cf869c2d82b

File: packages/js-recommendations/README.md (hunk 3 above)
Comment (shortcuts) [reply packages/js-recommendations/README.md]: Done in https://github.com/algolia/ui-components/pull/18/commits/522bcc398306cfdfa2cb264c40d32f69605f6812

File: packages/js-recommendations/README.md (hunk 2 above)
Comment (shortcuts) [reply packages/js-recommendations/README.md]: Removed abstraction in https://github.com/algolia/ui-components/pull/18/commits/1d53cbfc83fc5f8e4e981774917f08143849bb5a

**Review by francoischalifour — APPROVED** [review]

**Pull request merged**